### PR TITLE
Increase max weapon index for large modsets

### LINKED ISSUE #5042: Weapon on Back and Holster dont work All Weapons
**Arma 3 Version:** `1.68` and sometimes 1.66 stable
**CBA Version:** `3.2.1` stable
**ACE3 Version:** `3.9.1` stable

**Mods:**

- CBA_A3
- ace
RHS AFRF and USA
SMA 
and many Other. 
All Weapons...

**Description:**
Frist Time its some one with Arma version 1.66 and the Last RHS and ACE versions come. The Problem comes in the time not always. But since the Last Arma3 Update 1.68 Always. 
I and my Clan try to take our Rilfe on the Back. but thats dont work all have Standard Key Config. we Press 0 but nothing happend.

**Steps to reproduce:**
- Add the steps needed to reproduce the issue.
1. Take a Rifle
2. Take a Pistol
3. Take a binocular
Do you have the Rifle in youre hand you press 0 "Nothing work" do you change to the Pistol and press 0 nothing happend. If you have a binocular, you can change with B thats workes normal but sometimes if you press 0 he Change to the binocular. But the First steps are more impotante. 

**Where did the issue occur?**
Its happend on All

**Placed Modules:**
its happend with and without Moduls on ALL maps. 

**RPT log file:**
Nothing Errors in RPT and Debug to find. No more Infos about that to find i am sorry.

### PR DESCRIPTION
**Pull request by PabstMirror**
Fix #5042

### CODE REVIEW

**Comment by Dystopian** [comment]
#4978 doesn't agree

**Comment by commy2** [comment]
It's not just macros. Lots of duplicated code. Could use https://github.com/CBATeam/CBA_A3/blob/master/addons/common/fnc_selectWeapon.sqf
?

File: addons/respawn/functions/fnc_restoreGear.sqf [hunk 1]
```diff
@@ -44,7 +44,7 @@ if (!isNil "_activeWeaponAndMuzzle") then {
         private _index = 0;
 
         while {
-            _index < 100 && {currentWeaponMode _unit != _activeWeaponMode}
+            _index < 299 && {currentWeaponMode _unit != _activeWeaponMode}
```
Comment (dedmen) [inline addons/respawn/functions/fnc_restoreGear.sqf]: you added 200 to the other indexes. Why only add 199 here?

Comment (PabstMirror) [inline addons/respawn/functions/fnc_restoreGear.sqf]: Nothing special about the specific number, just needs to be bigger than biggest index.
This way they all use the same number `299`

**Pull request merged**